glusterd2/commands/global/setglobaloptions.go

```diff
@@ -0,0 +1,58 @@
+package globalcommands
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/cluster"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/store"
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gluster/glusterd2/pkg/errors"
+)
+
+func setGlobalOptionsHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	var req api.GlobalOptionReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusUnprocessableEntity, errors.ErrJSONParsingFailed.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	var c cluster.Cluster
+	resp, err := store.Store.Get(context.TODO(), cluster.ClusterPrefix)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrClusterNotFound.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	if resp.Count != 1 {
+		return
+	}
+
+	if err = json.Unmarshal(resp.Kvs[0].Value, &c); err != nil {
+		return
+	}
+
+	for k, v := range req.Options {
+		if _, found := cluster.GlobalOptMap[k]; found {
+			// TODO validate the value type for global option
```

Comment:
Is it possible to utilize the volume options validation functions here?

Reply:
The validation functions for volume options will work only on the separately defined option type in xlator/options/options.go. In fact Validate() is a method and works only on volume options for now. I do feel that they should be made reusable for cluster wide option validation.